### ISSUE TITLE
envelope() with fun=F3est

### ISSUE THREAD

**Issue by adchSTATS** [spatstat/spatstat#28]
When I use `envelope()` with `fun = F3est` and specify `rmax`, then the estimated F-functions unexpectedly "jump" at the very last r-value.
```
sim <- rpoispp3(lambda = 500)
F.env <- envelope(Y = sim, fun = F3est, rmax = 0.1)
plot.envelope(F.env, xlim = c(0, 0.1))
```
Spatstat version: 1.47-0.005
R version: 3.3.1

**Comment by rubak** [comment]
@baddstats is the real expert on this. Either it is a bug or it is related to the algorithm that calculates the estimate of F. At least in 2D the distance transform algorithm is used to estimate F in `Fest` and this imposes some requirements on the r-values that F is calculated for. See [these lines](https://github.com/spatstat/spatstat/blob/master/man/Fest.Rd#L174-L178) of the documentation for details.

If something similar is true for `F3est`, that could be the explanation, but I haven't looked at the details of the code. To circumvent the problem you can set `rmax = 0.1*128/127` and `nrval = 129` and then simply pick out the first 128 r-values. When I do this with your example the first 127 values `km` and `rs` are identical with the values obtained with `rmax=0.1`. For `cs` there is a small difference between the two results. In all cases the modification gives a correct looking 128th value (and then the omitted 129th value looks strange instead).

**Comment by baddstats** [comment]
This is a bug in the C code for `F3est`. It has now been fixed, in `spatstat 1.47-0.008`.
Thank you for bring this to our attention.
